src/main/java/com/example/bank/bankTransfer/transfer/history/export/XlsxTransferHistoryGenerator.java

```diff
@@ -0,0 +1,150 @@
+                cell.setCellValue(dateTime);
+                cell.setCellStyle(dateTimeCellStyle);
+                row.getSheet().setColumnWidth(i, DATE_COLUMN_WIDTH);
+            } else if (methodsForInsertingData.get(i) instanceof Long number) {
+                cell.setCellValue(number);
+            } else if (methodsForInsertingData.get(i) instanceof String name) {
+                cell.setCellValue(name);
+                cell.setCellStyle(defaultWrapTextStyle);
+            } else if (methodsForInsertingData.get(i) instanceof BigDecimal bigDecimal) {
+                cell.setCellValue(bigDecimal.doubleValue());
+                cell.setCellStyle(decimalCellStyle);
+            }
+        }
+    }
+
+    @NotNull
+    private static List<Object> createMethodsForInsertingDataIntoCells(TransferHistory history) {
+        return List.of(history.getCreatedOn(), history.getTransferType().toString(),
+                history.getExternalAccountNumber(), history.getTitle(), history.getAmount(), history.getBalance());
+    }
+
+    @NotNull
+    private CellStyle createStyleForHeader(Workbook workbook) {
+        CellStyle headerStyle = workbook.createCellStyle();
+        headerStyle.setFillForegroundColor(IndexedColors.GREY_25_PERCENT.getIndex());
+        headerStyle.setFillPattern(FillPatternType.SOLID_FOREGROUND);
+        headerStyle.setAlignment(HorizontalAlignment.CENTER);
+        Font font = workbook.createFont();
+        font.setBold(true);
+        headerStyle.setFont(font);
+        return headerStyle;
+    }
+
+    @NotNull
+    private static CellStyle createDefaultWrapTextStyle(Workbook workbook) {
+        CellStyle cellStyle = workbook.createCellStyle();
+        cellStyle.setWrapText(true);
+        return cellStyle;
+    }
+
+    @NotNull
+    private static CellStyle createDateTimeCellStyle(Workbook workbook) {
+        String dbFormatPattern = "YYYY-MMM-dd HH:mm:ss";
+        CellStyle style = workbook.createCellStyle();
+        style.setWrapText(true);
+        DataFormat dateTimeFormat = workbook.createDataFormat();
+        style.setDataFormat(dateTimeFormat.getFormat(dbFormatPattern));
+        return style;
+    }
+
+    @NotNull
+    private static CellStyle createDecimalCellStyle(Workbook workbook) {
+        String decimalFormatPattern = "#,##0.00";
+        CellStyle decimalStyle = workbook.createCellStyle();
+        DataFormat decimalFormat = workbook.createDataFormat();
+        decimalStyle.setDataFormat(decimalFormat.getFormat(decimalFormatPattern));
+        return decimalStyle;
+    }
+
+    private void saveWorkbookToFile(Workbook workbook, String accountNumber) {
+        String fileName = String.format(FILE_NAME_PATTERN, accountNumber);
+        try (FileOutputStream outputStream = new FileOutputStream(fileName)) {
+            workbook.write(outputStream);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Na gorze juz troche lepiej obsluzyles a tu owrapowales jeden wyjatek drugim i rzucasz. Musi byc chociaz jakis lepszy message a nie po prostu e ktory glownie zawiera stacktrace i nic po za tym. Fajine by bylo wiedziec deskryptywnie co poszlo nie tak. 
Chociazby : save workbook to file failed for i tu wstawic numer konta ktorego to dotyczy. 
Dzieki temu potem z logow ktos sie dowie ze ktos chcial wyjac jakis tam raport dla tego konta i mu sie nie udalo.